### PR TITLE
feat(billing.autorenew): add bulk action pages

### PR DESCRIPTION
**Pull request by marie-j**

### CODE REVIEW

File: client/app/billing/autoRenew/bulk/bulk.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                    data-ng-pluralize
+                    data-count="services.length"
+                    data-when="{
+                        '1': ('billing_autorenew_bulk_service' | translate),
+                        'other': ('billing_autorenew_bulk_services' | translate:{ count: services.length })
+                    }"
+                >
+                </strong>
+                <button
+                    class="oui-link"
```
Comment (antleblanc) [inline client/app/billing/autoRenew/bulk/bulk.html]: +`type="button"`

File: client/app/billing/autoRenew/bulk/translations/Messages_fr_FR.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "billing_autorenew_bulk": "Vous avez sélectionné {{ count }} services",
+  "billing_autorenew_bulk_cancel": "Annuler",
+  "billing_autorenew_bulk_confirm": "Valider",
+  "billing_autorenew_bulk_unavailable_pending_debt": "Les services ayant une facture en attente de réglèment",
```
Comment (antleblanc) [inline client/app/billing/autoRenew/bulk/translations/Messages_fr_FR.json]: :warning: Typo
```suggestion
    "billing_autorenew_bulk_unavailable_pending_debt": "Les services ayant une facture en attente de réglement",
```

File: client/app/billing/autoRenew/disable/disable.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        data-update-renew="$ctrl.update(services)">
+        <bulk-title>
+            <span data-translate="billing_autorenew_disable_title"></span>
+        </bulk-title>
+        <bulk-info-message>
+            <p>
+                <span class="d-block" data-translate="billing_autorenew_disable_description"></span>
+                <span class="d-block" data-translate="billing_autorenew_disable_warning"></span>
+            </p>
+            <p class="d-block" data-translate="billing_autorenew_disable_renew"></p>
```
Comment (antleblanc) [inline client/app/billing/autoRenew/disable/disable.html]: `<p>` element is already a block element.

```suggestion
            <p data-translate="billing_autorenew_disable_renew"></p>
```

File: client/app/billing/autoRenew/disable/disable.routing.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        BillingAutorenewEnable,
+        currentUser,
+        services,
+        servicesId,
+      ) => BillingAutorenewEnable.constructor.groupByAutorenewCapabilities(
+        _.filter(services, service => servicesId.includes((service.id).toString())),
+        currentUser.nichandle,
+      ),
+      updateRenew: /* @ngInject */
+          BillingAutoRenew => services => BillingAutoRenew.updateServices(
```
Comment (antleblanc) [inline client/app/billing/autoRenew/disable/disable.routing.js]: Indentation issue here?

File: client/app/billing/autoRenew/disable/translations/Messages_fr_FR.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "billing_autorenew_disable_title": "Désactiver le renouvellement automatique pour les services sélectionnés",
+  "billing_autorenew_disable_description": "Vous avez souhaité désactiver le renouvellement manuel pour les services sélectionnés.",
+  "billing_autorenew_disable_warning": "Dans ce mode de renouvellement, votre service n'est pas automatiquement reconduit lorsqu'il arrive à échéance.Plusieurs notifications vous seront envoyées par mails durant les semaines/jours précédents son terme, afin de vous prévenir de son expiration.",
```
Comment (antleblanc) [inline client/app/billing/autoRenew/disable/translations/Messages_fr_FR.json]:
```suggestion
  "billing_autorenew_disable_warning": "Dans ce mode de renouvellement, votre service n'est pas automatiquement reconduit lorsqu'il arrive à échéance. Plusieurs notifications vous seront envoyées par mails durant les semaines/jours précédents son terme, afin de vous prévenir de son expiration.",
```

File: client/app/billing/autoRenew/disable/translations/Messages_fr_FR.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+  "billing_autorenew_disable_renew": "Il vous suffira de renouveler votre service si vous souhaitez le conserver.",
+  "billing_autorenew_disable_data": "Attention, une fois le service expiré, il sera immédiatement suspendu, il ne sera plus accessible, et les données liées à ce service seront définitivement supprimées au bout de quelques jours sans règlement de votre part",
+  "billing_autorenew_disable_bulk_unavailable": "Le renouvellement automatique sera désactivé pour l'ensemble de vos services sélectionnés, exceptés pour  {{ count }} services :",
```
Comment (antleblanc) [inline client/app/billing/autoRenew/disable/translations/Messages_fr_FR.json]:
```suggestion
  "billing_autorenew_disable_bulk_unavailable": "Le renouvellement automatique sera désactivé pour l'ensemble de vos services sélectionnés, exceptés pour {{ count }} services :",
```

File: client/app/billing/autoRenew/bulk/bulk.controller.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+export default class {
+  $onInit() {
+    this.expand = {};
+
+    this.availableServices = this.services[undefined];
+    this.unavailableServices = _.omit(this.services, undefined);
```
Comment (FredericEspiau) [inline client/app/billing/autoRenew/bulk/bulk.controller.js]:
```suggestion
    this.unavailableServices = _.pickBy(this.services, service => service);
```

(`_.omit` will be removed in `Lodash 5`)

File: client/app/billing/autoRenew/bulk/bulk.controller.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+export default class {
+  $onInit() {
+    this.expand = {};
+
+    this.availableServices = this.services[undefined];
```
Comment (FredericEspiau) [inline client/app/billing/autoRenew/bulk/bulk.controller.js]: ?

File: client/app/billing/autoRenew/bulk/bulk.html [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+                        '1': ('billing_autorenew_bulk_service' | translate),
+                        'other': ('billing_autorenew_bulk_services' | translate:{ count: services.length })
+                    }"
+                >
+                </strong>
+                <button
+                    type="button"
+                    class="oui-link"
+                    data-ng-click="$ctrl.expand[reason] = ! $ctrl.expand[reason]"
+                    data-ng-bind="($ctrl.expand[reason] ? 'billing_autorenew_bulk_less' : 'billing_autorenew_bulk_more') | translate"></button>
```
Comment (antleblanc) [inline client/app/billing/autoRenew/bulk/bulk.html]: What do you think about adding a little bit of context of your `View more` link (#a11y concern)?

File: client/app/billing/autoRenew/bulk/bulk.controller.js (hunk 7 above)
Comment (FredericEspiau) [reply client/app/billing/autoRenew/bulk/bulk.controller.js]: Nevermind, you use `Lodash 3`, but keep it in mind :)

File: client/app/billing/autoRenew/bulk/bulk.html [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    </strong>
+    <p
+        data-translate="{{:: $ctrl.description }}"
+        data-translate-values="{
+            count: $ctrl.unavailableServicesCount
+        }">
+
+    </p>
+    <ul class="mb-4">
+        <li data-ng-repeat="(reason, services) in $ctrl.unavailableServices track by reason">
```
Comment (FredericEspiau) [inline client/app/billing/autoRenew/bulk/bulk.html]:
```suggestion
        <li data-ng-repeat="(reason, services) in ::$ctrl.unavailableServices track by reason">
```
?

File: client/app/billing/autoRenew/bulk/bulk.html [hunk 11]
```diff
@@ -0,0 +1,65 @@
+<oui-back-button data-on-click="$ctrl.goBack()">
```
Comment (FredericEspiau) [inline client/app/billing/autoRenew/bulk/bulk.html]:
```suggestion
<oui-back-button data-on-click=":: $ctrl.goBack()">
```
Everywhere  :)

File: client/app/billing/autoRenew/bulk/bulk.html [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+                    data-when="{
+                        '1': ('billing_autorenew_bulk_service' | translate),
+                        'other': ('billing_autorenew_bulk_services' | translate:{ count: services.length })
+                    }"
+                >
+                </strong>
+                <button
+                    type="button"
+                    class="oui-link"
+                    data-ng-click="$ctrl.expand[reason] = ! $ctrl.expand[reason]"
```
Comment (FredericEspiau) [inline client/app/billing/autoRenew/bulk/bulk.html]:
```suggestion
                    data-ng-click="$ctrl.expand[reason] = !$ctrl.expand[reason]"
```

File: client/app/billing/autoRenew/bulk/bulk.html [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+                    data-ng-bind="($ctrl.expand[reason] ? 'billing_autorenew_bulk_less' : 'billing_autorenew_bulk_more') | translate"></button>
+            </p>
+            <ul data-ng-if="$ctrl.expand[reason]">
+                <li
+                    data-ng-repeat="service in services track by service.id"
+                    data-ng-bind="service.domain">
+
```
Comment (FredericEspiau) [inline client/app/billing/autoRenew/bulk/bulk.html]: Extra whitespace

File: client/app/billing/autoRenew/bulk/bulk.html [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+                </strong>
+                <button
+                    type="button"
+                    class="oui-link"
+                    data-ng-click="$ctrl.expand[reason] = ! $ctrl.expand[reason]"
+                    data-ng-bind="($ctrl.expand[reason] ? 'billing_autorenew_bulk_less' : 'billing_autorenew_bulk_more') | translate"></button>
+            </p>
+            <ul data-ng-if="$ctrl.expand[reason]">
+                <li
+                    data-ng-repeat="service in services track by service.id"
```
Comment (FredericEspiau) [inline client/app/billing/autoRenew/bulk/bulk.html]:
```suggestion
                    data-ng-repeat="service in ::services track by service.id"
```
? (I don't know)

File: client/app/billing/autoRenew/bulk/translations/Messages_fr_FR.json [hunk 15]
```diff
@@ -0,0 +1,16 @@
+{
+  "billing_autorenew_bulk": "Vous avez sélectionné {{ count }} services",
+  "billing_autorenew_bulk_cancel": "Annuler",
+  "billing_autorenew_bulk_confirm": "Valider",
+  "billing_autorenew_bulk_unavailable_pending_debt": "Les services ayant une facture en attente de réglement",
+  "billing_autorenew_bulk_unavailable_already_automatic": "Les services bénéficiant déjà du paiement automatique",
+  "billing_autorenew_bulk_unavailable_one_shot_automatic": "Les services ne pouvant pas bénéficier du paiement automatique",
+  "billing_autorenew_bulk_unavailable_one_shot_manual": "Les services non compatible avec le renouvellement manuel ",
```
Comment (FredericEspiau) [inline client/app/billing/autoRenew/bulk/translations/Messages_fr_FR.json]:
```suggestion
  "billing_autorenew_bulk_unavailable_one_shot_manual": "Les services ne pouvant bénéficier du le renouvellement manuel ",
```
?

To have the same vocabulary everywhere

File: client/app/billing/autoRenew/bulk/translations/Messages_fr_FR.json [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+  "billing_autorenew_bulk_unavailable_nic_rights": "Les services pour lesquels vous n'êtes pas le contact facturation",
```
Comment (FredericEspiau) [inline client/app/billing/autoRenew/bulk/translations/Messages_fr_FR.json]:
```suggestion
  "billing_autorenew_bulk_unavailable_nic_rights": "Les services pour lesquels vous n'êtes pas le contact de facturation",
```
?

File: client/app/billing/autoRenew/bulk/bulk.html (hunk 9 above)
Comment (marie-j) [reply client/app/billing/autoRenew/bulk/bulk.html]: @antleblanc As seen together the sentence beforehand and the title adds context

**Pull request merged**